02week/pigLatin.js

```diff
@@ -10,11 +10,32 @@ const rl = readline.createInterface({
 
 function pigLatin(word) {
 
-  // Your code here
-
+  const translateWord = (word) => { // define translateWord with parameter word
+    const wordArray = word.toLowerCase().trim().split(''); // define an array: converts string to array and characters to lowercase and stores in array
+    const vowelsArray = ['a','e','i','o','u']; // define an array to hold all vowels for comparison
+    let wordArrayEnd = []; // define an array to store the last half of the word
+    let consonantArray = []; // define an array to store the first half of the word
+  
+    for (let step = 0; step < wordArray.length; step++) { // create a loop to step through each element in the word array
+      const letter = wordArray[step]; // define each value of the array as a letter
+      if (vowelsArray.includes(letter)) { // if the word array includes any items in the vowels array
+        if (step == [0]) { // and if the first letter in the array matches a vowel
+          return (word + 'yay'); // return the word with attached yay at the end 
+        } else { // otherwise
+          wordArrayEnd = wordArray.slice(step); // store the remaining elements in the array, introduced above, starting at the index of the first vowel found
```

Comment:
we want to try and avoid mutating data, so store the formatted word into a new const and use that

Reply:
how would i do that? just like wordArrayEndNew = wordArrayEnd?